spectral_cube/dask_spectral_cube.py

```diff
@@ -422,24 +486,47 @@ def apply_function_parallel_spectral(self,
             representing the mask.  It may also accept ``**kwargs``.  The
             function must return an object with the same shape as the input
             spectrum.
+        accepts_chunks : bool
+            Whether the function can take chunks with shape (ns, ny, nx) where
+            ``ns`` is the number of spectral channels in the cube and ``nx``
+            and ``ny`` may be greater than one.
+        save_to_tmp_dir : bool
+            If `True`, the computation will be carried out straight away and
+            saved to a temporary directory. This can improve performance,
+            especially if carrying out several operations sequentially. If
+            `False`, the computation is only carried out when accessing
+            specific parts of the data or writing to disk.
         kwargs : dict
             Passed to ``function``
         """
 
-        data = self._get_filled_data(fill=self._fill_value)
+        def wrapper(data, **kwargs):
+            if data.size > 0:
+                return function(data, **kwargs)
+            else:
+                return data
+
+        if accepts_chunks:
+
```

Comment:
this whitespace looks a bit off to me - not a big deal, but does this match the style used elsewhere?

Reply:
Fixed